src/middlewares/redirectMiddleware.ts

```diff
@@ -0,0 +1,47 @@
+import { NextRequest, NextResponse } from "next/server";
+
+import { readBaseUrl } from "src/app/env";
+import { HTTP_STATUSES } from "src/utils/http";
+import { RedirectDestinationSlugPage } from "studio/lib/interfaces/redirect";
+import { REDIRECT_BY_SOURCE_SLUG_QUERY } from "studio/lib/queries/redirects";
+
+export async function redirectMiddleware(
+  request: NextRequest,
+): Promise<NextResponse | undefined> {
+  const baseUrlResult = readBaseUrl();
+  if (!baseUrlResult.ok) {
+    console.error(
+      "Failed to run redirect middleware, missing base url:",
+      baseUrlResult.error,
+    );
+    return;
+  }
+  const baseUrl = baseUrlResult.value;
+  const slug = request.nextUrl.pathname;
+  // TODO: handle language path segment
```

Comment:
Should we add as github issue? As we're leaving the project

Reply:
Absolutely :+1: 

Redirect are kind of handled. Let's say that `/kalk` redirects to `/kalkulator`, and `/en/calculator` is a translation. Then `/en/kalk` will go to `/kalk` (assuming `/en/kalk` does not exist) which will then go to `/kalkulator` and finally `/en/calculator`. This is only ok if the user has the requested language as preferred, which is not necessarily true.

Created issue here: https://github.com/varianter/variant.no/issues/720

Reply:
👌